generate_stats.py

```diff
@@ -5,40 +5,65 @@
 sys.path.append('/opt/tljh/user/lib/moab/lib/python3.6/site-packages/pymoab-5.1.0-py3.6-linux-x86_64.egg')
 from pymoab import core, types
 from pymoab.rng import Range
-
+import matplotlib.pyplot as plt
+import numpy as np
 # import the new module that defines each of the functions
 import dagmc_stats
+import argparse
+
+
+
+
 
-def report_stats(entity_ranges):
+def report_stats(entity_ranges, surface_per_volume_stats, triangle_per_vertex_stats, verbose):
     """
     Method to print a table of statistics.
     
     inputs
     ------
     entity_ranges : a dictionary with one entry for each entity type that is a Range of handles to that type
     """
-
-    for entity_type, eh_range in entity_ranges.items():
-        print("There are {} entities of type {}.".format(eh_range.size(),entity_type))
+    if verbose:
+        for entity_type, eh_range in entity_ranges.items():
+            print("There are {} entities of type {}.".format(eh_range.size(),entity_type))
+        print('Type 0: Vertices \nType 2: Triangles \nType 11: EntitySets')
+        for stat, number in surface_per_volume_stats.items():
+            print('The {} number of Surfaces per Volume in this file is {}'.format(stat, number))
+        for stat, number in triangle_per_vertex_stats.items():
+            print('The {} number of Triangles per Vertex in this file is {}'.format(stat, number))
+    else:
+        for entity_type, eh_range in entity_ranges.items():
+            print("Type {}: {}".format(entity_type, eh_range.size()))
+        print("Surfaces per Volume:")
+        for stat, number in surface_per_volume_stats.items():
+            print('{}:{}'.format(stat, number))
+        print("Triangles per Vertex:")
+        for stat, number in triangle_per_vertex_stats.items():
+            print('{}:{}'.format(stat, number))
 
 def main():
 
     # starting with a single input file - will need to convert this to a user option
-    input_file = "3vols.h5m"
+    parser = argparse.ArgumentParser()
+    parser.add_argument("filename", help = "the file that you want read")
+    parser.add_argument("-v", "--verbose", action = "store_true", help = "increase output verbosity")
+    args = parser.parse_args()
+    input_file = args.filename
+    verbose = args.verbose
+    #input_file = "3vols.h5m"
```

Comment:
Feel free to delete lines instead of commenting them out - version Control (I.e. git) will always let you get things back if you decide you need them in the future